### ISSUE TITLE
Notion - When updating contents of existing page , new page gets created with the updated contents.

### ISSUE THREAD

**Issue by sangee2004** [obot-platform/obot#251]
**Steps to reproduce the problem:**
1. Create a agent with all tools from Notion.
2. Ask to create a new page 
```
Create a page "MyNew" with heading "my task list" and todo task "hello" under "My new test page"
```
New page gets created as expected.

3. Ask to update this page with some updated contents
```
Add "my task list updated " and todo task "hello updated" in "MyNew" page
```

This results in another page with same name being created.
<img width="266" alt="Screenshot 2024-10-21 at 12 47 57 PM" src="https://github.com/user-attachments/assets/ffa22eee-27ae-4239-ab52-e1d4059ea1de">

The new page gets created with following contents

<img width="621" alt="Screenshot 2024-10-21 at 12 47 09 PM" src="https://github.com/user-attachments/assets/c4aee942-a2b4-4471-9bdb-338e9859399f">

**Expected Behavior:**
Existing page should be updated instead.

Stack trace -[calls (24).json](https://github.com/user-attachments/files/17466373/calls.24.json)

**Comment by sangee2004** [comment]
@njhale I am able to get update to result in creation of new files when testing with latest builds:

```
  "github.com/otto8-ai/tools": "9abc7da7146cbd8448dca8ed11fde418482f2341",
  "otto": "v0.0.0-dev+b1a78bfe"
```

**Steps that were followed:**

1. Ask to `create a new page "testingupdatenow" with text "hello123"` and when asked to provide the parent page  asked to- `Create it under "My new test page"`
This page gets created successfully.
2. Ask to `Update this page and add content -"The end"`

This results in new page being created  with the updates.

![Image](https://github.com/user-attachments/assets/a9cf8a20-148e-40be-8a12-99254a7dba36)

Debug logs - [calls - 2024-11-18T164221.495.json](https://github.com/user-attachments/files/17807738/calls.-.2024-11-18T164221.495.json)

3. Ask to `update content of "kitten" and add "the end"` ( `kitten` is a page that already exists)

This results in 3 new pages getting created because of the page content being large.

![Image](https://github.com/user-attachments/assets/98d4b136-8dc2-4020-830f-22f9e7627c3a)
![Image](https://github.com/user-attachments/assets/bbaeca55-1841-4bda-b4d5-bd230166de03)

Debug logs - [calls - 2024-11-18T164452.475.json](https://github.com/user-attachments/files/17807757/calls.-.2024-11-18T164452.475.json)


From Notion that shows the new files created:
![Image](https://github.com/user-attachments/assets/7f14f7b7-866c-4324-93ec-3d0fa23fb6a3)

**Comment by sangee2004** [comment]
@thedadams No. This one tracks the need to not allow for  chats relating to update result in creating new notion pages. #627 is to track that we are not able to create new pages when contents are large.

**Comment by njhale** [comment]
OK, at this point we're on the 3rd iteration of "NEVER DO THAT" in the notion context tool. It seems like no amount of coaxing in the context is going to prevent the LLM from making this mistake.

With that in mind, I think the only way to "correct" this is by implementing an `Update Page` tool.

Later this morning I'll open a dedicated enhancement issue for that and make sure this issue is tracked along side it.

**Comment by tybalex** [comment]
@sangee2004 PR#267 https://github.com/otto8-ai/tools/pull/267 introduces a new Notion tool for updating contents of existing page, which should address this issue.

**Comment by cjellick** [comment]
@tybalex when your pr merges, you should move the issue to testing. i have done it for this one

**Comment by sangee2004** [comment]
@tybalex I am testing with latest build 
```
 "github.com/otto8-ai/tools": "4c068dc65cb743bf469bb8a0378a01d4351cb51f",
  "otto": "v0.0.0-dev+32b70a24"
```
Scenario 1:

When I tried to update the contents of the page I created , i dont see "Update Page" tool being called.

<img width="643" alt="Image" src="https://github.com/user-attachments/assets/8c129f5c-fadd-4feb-8184-61269b4d877b" />

debug logs -  [calls (46).json](https://github.com/user-attachments/files/18133146/calls.46.json)

Scenario 2:


When I tried to update the contents of the page I created , i see "Create Page" being called,


<img width="468" alt="Image" src="https://github.com/user-attachments/assets/b5c23721-19e1-4b1c-98b0-dafcfe24754e" />


debug logs - [calls (45).json](https://github.com/user-attachments/files/18133145/calls.45.json)

**Comment by cjellick** [comment]
@sangee2004 i looked at your debug json. It does not have the update page tool. Please retest with the updatePage tool added

**Comment by sangee2004** [comment]
@cjellick I am testing with the latest version of tool and see the same behavior where update page tool is not being called.

I do see the `update page` tool in the tool list in the debug logs attached here - https://github.com/acorn-io/acorn/issues/251#issuecomment-2542598567

```
   "tool": {
      "id": "/otto8-tools/notion/tool.gpt:Notion Context",
      "instructions": "#!sys.echo\n\n## Instructions for using Notion tools\n\nYou have access to a set of tools to interact with Notion.\nYou cannot use the Create Page tool to update the contents of a page. It is only for creating new pages.\nEvery time you are about to create a new page and the user isn't explicitly asking for one, ask them to confirm before proceeding.\n\n## End of instructions for using Notion tools",
      "internalPrompt": null,
      "localTools": {
        "add database row": "/otto8-tools/notion/tool.gpt:Add Database Row",
        "create page": "/otto8-tools/notion/tool.gpt:Create Page",
        "get database": "/otto8-tools/notion/tool.gpt:Get Database",
        "get database properties": "/otto8-tools/notion/tool.gpt:Get Database Properties",
        "get page": "/otto8-tools/notion/tool.gpt:Get Page",
        "list users": "/otto8-tools/notion/tool.gpt:List Users",
        "notion": "/otto8-tools/notion/tool.gpt:Notion",
        "notion context": "/otto8-tools/notion/tool.gpt:Notion Context",
        "search": "/otto8-tools/notion/tool.gpt:Search",
        "update database row": "/otto8-tools/notion/tool.gpt:Update Database Row",
        "update page": "/otto8-tools/notion/tool.gpt:Update Page"
      },
```
Is there an additional place were I need to check in the debug logs for `Update Page` tool to be present ?

**Comment by tybalex** [comment]
@sangee2004 yes. something weird here: from line 149 to line 308 `the tools and tool-mappings` of the actual chat completion -- the `Update Page` tool is not listed in neither of them

**Comment by tybalex** [comment]
Add another minor fix -- add `Update Page` to the Notion tool bundle. Sangeetha has been using the Notion bundle for testing, while I was using the individual tools. This PR will likely fix the issue.

**Comment by cjellick** [comment]
Status of this one may have gotten lost when i created a new project board. I think it should be back to test

**Comment by tybalex** [comment]
@sangee2004 can you confirm the latest version has fixed this issue when you got some time?

**Comment by sangee2004** [comment]
@tybalex I am seeing this issue addressed when testing with latest version of the tool.
```
github.com/otto8-ai/tools:

43e132b64079e5e90ff70c79d0c0c33dcbf25e74
```
Updating contents of existing page works as expected now.
